R/multitables.R

```diff
@@ -85,25 +85,32 @@ setMethod("is.public<-", "Multitable", function (x, value) {
 #' name(m) # [1] "gender + age4 + marstat"
 #' }
 #' @export
-newMultitable <- function (formula, data, name, ...) {
-    ## Validate inputs
-    if (missing(formula)) {
-        halt("Must provide a formula")
-    }
-    if (missing(data) || !is.dataset(data)) {
-        halt(dQuote("data"), " must be a Dataset")
-    }
+newMultitable <- function (formula, data, name, multitable=NULL, ...) {
+
+    if (missing(multitable)){
+        ## Validate inputs
+        if (missing(formula)) {
+            halt("Must provide a formula")
+        }
+        if (missing(data) || !is.dataset(data)) {
+            halt(dQuote("data"), " must be a Dataset")
+        }
+
+        template <- formulaToQuery(formula, data)
+        if (missing(name)) {
+            name <- formulaRHS(formula)
+        }
 
-    template <- formulaToQuery(formula, data)
-    if (missing(name)) {
-        name <- formulaRHS(formula)
+        payload <- wrapEntity(
+            name=name,
+            template=lapply(template$dimensions,
+                function (x) list(query=x, variable=findVariableReferences(x)))
+        )
+    } else {
+        payload = wrapEntity(name=name,
+                             multitable=self(multitable))
     }
 
-    payload <- wrapEntity(
-        name=name,
-        template=lapply(template$dimensions,
-            function (x) list(query=x, variable=findVariableReferences(x)))
-    )
     ## TODO: remove "variable" from that--no longer required?
```

Comment:
While you're in here, how about this TODO? I think this "variable" thing was cleaned up from the API a while ago.